core/corehttp/gateway_handler.go

```diff
@@ -55,16 +55,20 @@ func (i *gatewayHandler) newDagFromReader(r io.Reader) (*dag.Node, error) {
 
 // TODO(btc): break this apart into separate handlers using a more expressive muxer
 func (i *gatewayHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithCancel(i.node.Context())
+	defer cancel()
+
 	if i.config.Writable {
 		switch r.Method {
 		case "POST":
-			i.postHandler(w, r)
+			i.postHandler(ctx, w, r)
 			return
 		case "PUT":
-			i.putHandler(w, r)
+			// TODO(cryptix): where are the docs?
+			http.Error(w, "writableGateway: PUT method not meaningful on IPFS - use POST and see the docs", http.StatusMethodNotAllowed)
```

Comment:
Can just remove the PUT-case and let it fall through to the method error in lines 81 ff.